src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1206,13 +1274,106 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
   // Register field_name in the blueprint_index group.
   if(myid == 0)
   {
-    RegisterFieldInBPIndex(field_name, gf);
+    // Note: The bp index requires GridFunction::VectorDim()
+    //       since the GF might be scalar valued and have a vector basis
+    //       (e.g. hdiv and hcurl spaces)
+    RegisterFieldInBPIndex(field_name, gf->VectorDim());
   }
 
   // Register field_name + gf in field_map.
   DataCollection::RegisterField(field_name, gf);
 }
 
+void MFEMSidreDataCollection::RegisterQField(const std::string& field_name,
+                                             mfem::QuadratureFunction* qf,
+                                             const std::string& buffer_name,
+                                             axom::sidre::IndexType offset)
+{
+  if(field_name.empty() || buffer_name.empty() || qf == NULL ||
+     qf->GetSpace() == NULL)
+  {
+  #ifdef AXOM_DEBUG
```

Comment:
This message should be split into a few cases since the current message doesn't adequately describe what happened in all cases.

Reply:
This check is analogous to this existing snippet for GridFunctions (with the exception of NULL vs nullptr):

https://github.com/LLNL/axom/blob/614075596fe967d574632b3a0cb1be525ff3edd7/src/axom/sidre/core/MFEMSidreDataCollection.cpp#L1202-L1206

The above case doesn't even provide a message, so I'll go ahead and put some more helpful messages in both places